kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/CreateExternalOrderCall.java

```diff
@@ -51,19 +58,13 @@ public void run() {
 				String orderID = extractOrderID(e.getResponseHeaders());
 				getOrder(orderID);
 			} else {
-
-				runOnMainThread(new Runnable() {
-					@Override
-					public void run() {
-						externalOrderCallbacks.onOrderFailed(ErrorUtil.fromApiException(e));
-					}
-				});
+				sendOrderCreationFailedEvent(e);
+				onOrderFailed(ErrorUtil.fromApiException(e));
```

Comment:
we will not have information regarding the order - if we arrived to the catch it means the openOrder is null